frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx

```diff
@@ -216,6 +266,90 @@ export const EditDate = ({ field }: EditDateProps): JSX.Element => {
           {get(errors, 'dateValidation.customMinDate.message')}
         </FormErrorMessage>
       </FormControl>
+      <Stack>
+        <FormControl isReadOnly={isLoading}>
+          <Toggle
+            {...register('restrictParticularDays.addParticularDayRestriction')}
+            label="Customize days of the week"
+          />
+          <Text textStyle="body-2" color="secondary.400">
+            Checking a day will disable all the same days in the calendar
+          </Text>
+        </FormControl>
+        {getValues('restrictParticularDays.addParticularDayRestriction') ? (
+          <FormControl isRequired isReadOnly={isLoading}>
+            <Stack direction="row" spacing="0.75rem">
+              <Stack direction="column">
+                {PARTICULAR_DAYS_OPTIONS_LEFT.map((option) => {
+                  return (
+                    <Box width="9.25rem" key={option.index}>
+                      <Controller
+                        name="restrictParticularDays.invalidDaysOfTheWeek"
+                        control={control}
+                        key={option.index}
+                        render={({ field: { onChange, value, ref } }) => {
+                          return (
+                            <Checkbox
+                              onChange={() => {
+                                const idx = value.indexOf(option.index)
+                                value =
+                                  idx > -1
+                                    ? value.filter(
+                                        (dayId) => dayId !== option.index,
+                                      )
+                                    : [...value, option.index]
+                                onChange(value)
+                              }}
+                              ref={ref}
+                              isChecked={value.includes(option.index)}
+                            >
+                              {option.title}
+                            </Checkbox>
+                          )
+                        }}
+                      />
+                    </Box>
+                  )
+                })}
```

Comment:
probably can use it as an uncontrolled component:

```tsx
invalidCheckboxValues.map((o, idx) => (
        <Checkbox
          value={o}
          {...register(checkboxInputName, validationRules)}
        >
          {o}
        </Checkbox>
```

Reply:
As discussed previously, a controlled checkbox group component had to be used to store the checkbox values in an array.